<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>官文亮</author>
 <class>SerialTool</class>
 <widget class="QMainWindow" name="SerialTool">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>426</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string notr="true">SerialTool</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resource/serialtool.qrc">
    <normaloff>:/SerialTool/images/icon.ico</normaloff>:/SerialTool/images/icon.ico</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>16777215</height>
    </size>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="spacing">
     <number>3</number>
    </property>
    <property name="leftMargin">
     <number>1</number>
    </property>
    <property name="topMargin">
     <number>1</number>
    </property>
    <property name="rightMargin">
     <number>1</number>
    </property>
    <property name="bottomMargin">
     <number>1</number>
    </property>
    <item>
     <widget class="QWidget" name="portConfigWidget" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="leftMargin">
        <number>2</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Port</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxPortNum">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>120</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>120</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">QComboBox { width: 120px }</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Baud Rate</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxBaudRate">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">QComboBox { width: 80px }</string>
         </property>
         <property name="currentText">
          <string notr="true">1200</string>
         </property>
         <property name="maxVisibleItems">
          <number>20</number>
         </property>
         <item>
          <property name="text">
           <string notr="true">1200</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">2400</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">4800</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">9600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">19200</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">38400</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">57600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">115200</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">128000</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">230400</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">256000</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">345600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">460800</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">576000</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">921600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">1843200</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabTester">
       <attribute name="title">
        <string>Tester</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>2</number>
        </property>
        <property name="leftMargin">
         <number>2</number>
        </property>
        <property name="topMargin">
         <number>2</number>
        </property>
        <property name="rightMargin">
         <number>2</number>
        </property>
        <property name="bottomMargin">
         <number>2</number>
        </property>
        <item>
         <widget class="QSplitter" name="splitter">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="TextEdit" name="textEditRx" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
          </widget>
          <widget class="QWidget" name="layoutWidget">
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item alignment="Qt::AlignBottom">
             <widget class="QWidget" name="widget_1" native="true">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_4">
               <property name="spacing">
                <number>3</number>
               </property>
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QGroupBox" name="groupBox">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="title">
                  <string>Rx Settings</string>
                 </property>
                 <layout class="QVBoxLayout" name="verticalLayout_5">
                  <property name="spacing">
                   <number>2</number>
                  </property>
                  <property name="leftMargin">
                   <number>4</number>
                  </property>
                  <property name="rightMargin">
                   <number>4</number>
                  </property>
                  <property name="bottomMargin">
                   <number>4</number>
                  </property>
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_3">
                    <item>
                     <widget class="QRadioButton" name="portReadAscii">
                      <property name="text">
                       <string notr="true">ASCII</string>
                      </property>
                      <property name="checked">
                       <bool>true</bool>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QRadioButton" name="portReadHex">
                      <property name="text">
                       <string notr="true">Hex</string>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                  <item>
                   <widget class="QCheckBox" name="wrapLineBox">
                    <property name="enabled">
                     <bool>true</bool>
                    </property>
                    <property name="text">
                     <string>Wrap Line</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
               <item>
                <widget class="QGroupBox" name="groupBox_2">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="title">
                  <string>Tx Settings</string>
                 </property>
                 <layout class="QVBoxLayout" name="verticalLayout_6">
                  <property name="spacing">
                   <number>2</number>
                  </property>
                  <property name="leftMargin">
                   <number>4</number>
                  </property>
                  <property name="rightMargin">
                   <number>4</number>
                  </property>
                  <property name="bottomMargin">
                   <number>4</number>
                  </property>
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_4">
                    <item>
                     <widget class="QRadioButton" name="portWriteAscii">
                      <property name="text">
                       <string notr="true">ASCII</string>
                      </property>
                      <property name="checked">
                       <bool>true</bool>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QRadioButton" name="portWriteHex">
                      <property name="text">
                       <string notr="true">Hex</string>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_6">
                    <property name="spacing">
                     <number>2</number>
                    </property>
                    <item>
                     <widget class="QCheckBox" name="resendBox">
                      <property name="text">
                       <string>Resend</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QSpinBox" name="spinBoxStepTime">
                      <property name="minimum">
                       <number>10</number>
                      </property>
                      <property name="maximum">
                       <number>120000</number>
                      </property>
                      <property name="value">
                       <number>1000</number>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLabel" name="label_3">
                      <property name="text">
                       <string>ms</string>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_3">
              <property name="verticalSpacing">
               <number>3</number>
              </property>
              <item row="2" column="1">
               <widget class="QPushButton" name="sendButton">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>Ctrl+Enter</string>
                </property>
                <property name="text">
                 <string>Send</string>
                </property>
                <property name="shortcut">
                 <string>Ctrl+Return</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QComboBox" name="comboBox"/>
              </item>
              <item row="0" column="0" colspan="2">
               <widget class="TextEdit" name="textEditTx" native="true">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Ignored">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabOsc">
       <attribute name="title">
        <string>Plot</string>
       </attribute>
       <widget class="Oscilloscope" name="oscPlot" native="true">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>592</width>
          <height>297</height>
         </rect>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tabFileTrans">
       <attribute name="title">
        <string>File Transmit</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="FileTransferView" name="fileTransfer" native="true"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>600</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionOption"/>
    <addaction name="actionVedioBox"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="portSwitchAction"/>
    <addaction name="portRunAction"/>
    <addaction name="clearAction"/>
    <addaction name="separator"/>
    <addaction name="portSetAction"/>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionVisibleTab0"/>
    <addaction name="actionVisibleTab1"/>
    <addaction name="separator"/>
    <addaction name="actionVisibleToolbar"/>
    <addaction name="actionVisibleStatusBar"/>
   </widget>
   <widget class="QMenu" name="menu_5">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_3"/>
   <addaction name="menu_4"/>
   <addaction name="menu_2"/>
   <addaction name="menu_5"/>
  </widget>
  <widget class="QToolBar" name="toolBar1">
   <property name="windowTitle">
    <string>Tool Bar</string>
   </property>
   <property name="allowedAreas">
    <set>Qt::BottomToolBarArea|Qt::TopToolBarArea</set>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="portSetAction"/>
   <addaction name="separator"/>
   <addaction name="portSwitchAction"/>
   <addaction name="portRunAction"/>
   <addaction name="clearAction"/>
   <addaction name="separator"/>
   <addaction name="actionOption"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../resource/serialtool.qrc">
     <normaloff>:/SerialTool/images/save.ico</normaloff>:/SerialTool/images/save.ico</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionOption">
   <property name="icon">
    <iconset resource="../resource/serialtool.qrc">
     <normaloff>:/SerialTool/images/config.ico</normaloff>:/SerialTool/images/config.ico</iconset>
   </property>
   <property name="text">
    <string>Options</string>
   </property>
  </action>
  <action name="portSwitchAction">
   <property name="icon">
    <iconset resource="../resource/serialtool.qrc">
     <normaloff>:/SerialTool/images/connect.png</normaloff>:/SerialTool/images/connect.png</iconset>
   </property>
   <property name="text">
    <string>Open Port</string>
   </property>
   <property name="shortcut">
    <string>Alt+S</string>
   </property>
  </action>
  <action name="portRunAction">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../resource/serialtool.qrc">
     <normaloff>:/SerialTool/images/pause.ico</normaloff>
     <activeoff>:/SerialTool/images/pause.png</activeoff>:/SerialTool/images/pause.ico</iconset>
   </property>
   <property name="text">
    <string>Pause Tx/Rx</string>
   </property>
   <property name="shortcut">
    <string>Alt+R</string>
   </property>
  </action>
  <action name="clearAction">
   <property name="icon">
    <iconset resource="../resource/serialtool.qrc">
     <normaloff>:/SerialTool/images/clear.png</normaloff>:/SerialTool/images/clear.png</iconset>
   </property>
   <property name="text">
    <string>Clear Buffer</string>
   </property>
   <property name="shortcut">
    <string>Alt+C</string>
   </property>
  </action>
  <action name="portSetAction">
   <property name="icon">
    <iconset resource="../resource/serialtool.qrc">
     <normaloff>:/SerialTool/images/port config.png</normaloff>:/SerialTool/images/port config.png</iconset>
   </property>
   <property name="text">
    <string>Port Settings</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="icon">
    <iconset resource="../resource/serialtool.qrc">
     <normaloff>:/SerialTool/images/exit.ico</normaloff>:/SerialTool/images/exit.ico</iconset>
   </property>
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="actionVisibleToolbar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Tool Bar</string>
   </property>
  </action>
  <action name="actionVisibleTab0">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Tester</string>
   </property>
   <property name="toolTip">
    <string>Tester</string>
   </property>
  </action>
  <action name="actionVisibleTab1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Plot</string>
   </property>
  </action>
  <action name="actionVisibleStatusBar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Status Bar</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionVedioBox">
   <property name="text">
    <string>Vedio Box</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>TextEdit</class>
   <extends>QWidget</extends>
   <header>textedit.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Oscilloscope</class>
   <extends>QWidget</extends>
   <header>oscilloscope.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FileTransferView</class>
   <extends>QWidget</extends>
   <header>filetransferview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resource/serialtool.qrc"/>
 </resources>
 <connections/>
</ui>
